datafusion/core/tests/user_defined/user_defined_plan.rs

```diff
@@ -295,20 +296,175 @@ async fn topk_plan() -> Result<()> {
     Ok(())
 }
 
+#[tokio::test]
+/// Run invariant checks on the logical plan extension [`TopKPlanNode`].
+async fn topk_invariants() -> Result<()> {
```

Comment:
test: demonstrate it fails the user-defined invariant for the created logical plan.